### ISSUE TITLE
I need to figure out how to setup the business logic of the backend (needs to be done before moving to auth)

### ISSUE THREAD

**Issue by Ali-Kabbadj** [Ali-Kabbadj/khmissa-shake#7]
I have plans for a lot of features, and i need a robust business logic on the backend side to make the project manageable and structured in an understandable way.

**Comment by Ali-Kabbadj** [comment]
I already read about DDD before, but i never implemented it, i did some more research now, and it seems to be a good structure for the kind of project i am trying to build here.

i could use something like feature modules too, to separate each major feature from the other

I will get back to this once i found a suitable structure.

**Comment by Ali-Kabbadj** [comment]
So, i did some more research and i think i am going to go with a microservices approach rather that a monolithic architecture, i will create a separate issue for the brainstorming a concise solution.
